io/net/bonding.py

```diff
@@ -157,29 +162,57 @@ def ping_check(self, arg1):
         cmd = "ping -I %s %s -c 5"\
               % (self.bond_name, self.peer_first_ipinterface)
         if process.system(cmd, shell=True, ignore_status=True) != 0:
-            self.fail("ping failed in Mode %s, check bonding configuration"
-                      % arg1)
+            return False
+        return True
 
     def bond_fail(self, arg1):
         '''
         bond fail
         '''
+        if len(self.host_interfaces) > 1:
+            for interface in self.host_interfaces:
+                self.log.info("Failing interface %s for mode %s"
+                              % (interface, arg1))
+                cmd = "ifconfig %s down" % interface
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.fail("bonding not working when trying to down the\
+                               interface %s " % interface)
+                time.sleep(self.sleep_time)
+                if self.ping_check(arg1):
+                    self.log.info("Ping passed for Mode %s" % arg1)
+                else:
+                    self.fail("ping failed in Mode %s, check \
+                               bonding configuration" % arg1)
+                process.system_output(self.bond_status, shell=True,
+                                      verbose=True)
+                cmd = "ifconfig %s up" % interface
+                time.sleep(self.sleep_time)
+                if process.system(cmd, shell=True, ignore_status=True) != 0:
+                    self.fail("Not able to bring up the slave\
+                                    interface %s" % interface)
+                time.sleep(self.sleep_time)
+        else:
+            self.log.warning("Need a min of 2 host interfaces to test\
```

Comment:
I agree. It should be as "Debug" otherwise it does not look good that the test shows a warning even when it has passed.